src/hal/lpc11xx/syscon.rs

```diff
@@ -0,0 +1,199 @@
+  match peripheral {
+    ResetPeripheral::SPI0 => {
+      regs::SYSCON().presetctrl.set_ssp0_rst_n(
+        regs::SYSCON_presetctrl_ssp0_rst_n::SPIO0RESET);
+      regs::SYSCON().presetctrl.set_ssp0_rst_n(
+        regs::SYSCON_presetctrl_ssp0_rst_n::SPIO0NORESET);
+    },
+    ResetPeripheral::SPI1 => {
+      regs::SYSCON().presetctrl.set_ssp1_rst_n(
+        regs::SYSCON_presetctrl_ssp1_rst_n::SPI1RESET);
+      regs::SYSCON().presetctrl.set_ssp1_rst_n(
+        regs::SYSCON_presetctrl_ssp1_rst_n::SPI2NORESET);
+    },
+    ResetPeripheral::I2C => {
+      regs::SYSCON().presetctrl.set_i2c_rst_n(
+        regs::SYSCON_presetctrl_i2c_rst_n::I2CRESET);
+      regs::SYSCON().presetctrl.set_i2c_rst_n(
+        regs::SYSCON_presetctrl_i2c_rst_n::I2CNORESET);
+    },
+    ResetPeripheral::CAN => {
+      regs::SYSCON().presetctrl.set_can_rst_n(
+        regs::SYSCON_presetctrl_can_rst_n::CANRESET);
+      regs::SYSCON().presetctrl.set_can_rst_n(
+        regs::SYSCON_presetctrl_can_rst_n::CANNORESET);
+    }
+  }
+}
+
+pub fn init_system_clock() {
+  regs::SYSCON().pdruncfg
+      .set_sysosc_pd(regs::SYSCON_pdruncfg_sysosc_pd::POWERED);
+  regs::SYSCON().sysoscctrl.ignoring_state()
+      .set_bypass(regs::SYSCON_sysoscctrl_bypass::NOBYPASS)
+      .set_freqrange(regs::SYSCON_sysoscctrl_freqrange::LOW);
+  regs::SYSCON().syspllclksel.ignoring_state()
+      .set_sel(regs::SYSCON_syspllclksel_sel::SYSTEM_OSCILLATOR);
+
+  regs::SYSCON().syspllclkuen.ignoring_state()
+      .set_ena(regs::SYSCON_syspllclkuen_ena::UPDATE_CLOCK_SOURCE);
+  regs::SYSCON().syspllclkuen.ignoring_state()
+      .set_ena(regs::SYSCON_syspllclkuen_ena::NO_CHANGE);
+  regs::SYSCON().syspllclkuen.ignoring_state()
+      .set_ena(regs::SYSCON_syspllclkuen_ena::UPDATE_CLOCK_SOURCE);
+
+  loop {
+    if regs::SYSCON().syspllclkuen.ena() == regs::SYSCON_syspllclkuen_ena::UPDATE_CLOCK_SOURCE {
+      break
+    }
+  }
+}
+
+#[cfg(test)]
+mod test {
+  use super::*;
+  use volatile_cell::{VolatileCellReplayer, set_replayer};
+  use expectest::prelude::*;
+  use expectest;
+  use std::thread;
+  use std::string::String;
+  use std::convert::From;
+
+  #[test]
+  fn returns_isr_location() {
+    init_replayer!();
+
+    expect_volatile_read!(0x4004_8000, 0b10);
+
+    expect!(get_isr_location()).to(be_equal_to(ISRLocation::Flash));
+
+    expect_replayer_valid!();
+  }
+
+  #[test]
+  fn sets_isr_location() {
+    init_replayer!();
+
+    expect_volatile_write!(0x4004_8000, 2);
+
+    set_isr_location(ISRLocation::Flash);
+
+    expect_replayer_valid!();
+  }
+
+  #[test]
+  fn fails_to_set_isr_location_to_bootloader() {
+    let j = thread::Builder::new()
+        .name(String::from("fails_to_set_isr_location_to_bootloader"))
+        .spawn(|| {
+      init_replayer!();
+      expect_volatile_write!(0x4004_8000, 0);
+      set_isr_location(ISRLocation::Bootloader);
+    }).unwrap();
+    let res = j.join();
+
+    expect!(res.is_err()).to(be_equal_to(true));
```

Comment:
> Two hip-hip hoorays for having to track nightlies!

Did I tell you that we will need to fork cargo to keep the nightly going? Buy the ticket, take the ride! :roller_coaster: